src/EditorFeatures/CSharp/EventHookup/EventHookupCommandHandler_TabKeyCommand.cs

```diff
@@ -181,12 +181,12 @@ private void GenerateAndAddEventHandler(ITextView textView, ITextBuffer subjectB
             var simplifiedDocument = Simplifier.ReduceAsync(documentWithNameAndAnnotationsAdded.WithSyntaxRoot(updatedRoot), Simplifier.Annotation, cancellationToken: cancellationToken).WaitAndGetResult(cancellationToken);
             var formattedDocument = Formatter.FormatAsync(simplifiedDocument, Formatter.Annotation, cancellationToken: cancellationToken).WaitAndGetResult(cancellationToken);
 
-            plusEqualTokenEndPosition = formattedDocument
-                .GetSyntaxRootAsync(cancellationToken).WaitAndGetResult(cancellationToken)
-                .GetAnnotatedNodesAndTokens(plusEqualsTokenAnnotation)
-                .Single().Span.End;
+            var newRoot = formattedDocument.GetSyntaxRootSynchronously(cancellationToken);
+            plusEqualTokenEndPosition = newRoot.GetAnnotatedNodesAndTokens(plusEqualsTokenAnnotation)
+                                               .Single().Span.End;
 
-            return document.Project.Solution.WithDocumentText(formattedDocument.Id, formattedDocument.GetTextAsync(cancellationToken).WaitAndGetResult(cancellationToken));
+            return document.Project.Solution.WithDocumentText(
+                formattedDocument.Id, formattedDocument.GetTextAsync(cancellationToken).WaitAndGetResult(cancellationToken));
```

Comment:
Should we have a similar path for GetTextSynchronously here as well? Otherwise not sure if this change here really gets us anything.